src/schema.ts

```diff
@@ -247,19 +273,32 @@ function timeSummary(timeunit: TimeUnit, summary: Summary): Summary {
 /**
  * @return a new unique object based off of the old unique count and a binning scheme
  */
-function binUnique(bin, oldUnique) {
+function binUnique(bin, oldUnique, excludeInvalid: boolean) {
   const newUnique = {};
   for (var value in oldUnique) {
     let bucket: number = bin.value(Number(value)) as number;
-    if (!newUnique[bucket]) {
-      newUnique[bucket] = oldUnique[value];
-    } else {
-      newUnique[bucket] += oldUnique[value];
+    if (!excludeInvalid || (bucket !== null && bucket === bucket)) {
+      if (!newUnique[bucket]) {
+        newUnique[bucket] = oldUnique[value];
```

Comment:
`newUnique[bucket] = (newUnique[bucket] || 0) + oldUnique[value];`  is more concise.  :)